### ISSUE TITLE
Install not working

### ISSUE THREAD

**Issue by carlybatist** [shyamblast/Koogu#1]
I am trying to install but pip install isn't working correctly - I pip install and it runs that all requirements have been satisfied but then the imports don't work (see attached screenshots) - thoughts on why this is happening?

<img width="899" alt="Screen Shot 2022-08-13 at 7 40 41 AM" src="https://user-images.githubusercontent.com/46716228/184499053-ddcf1a50-0bb6-4eb8-b48c-f7d01a2418ef.png">

<img width="794" alt="Screen Shot 2022-08-13 at 7 40 49 AM" src="https://user-images.githubusercontent.com/46716228/184499042-d9d2227b-a4c7-4bf9-8dc0-b58dc271b11c.png">

**Comment by shyamblast** [comment]
This could be an anaconda specific issue (I haven't tried koogu on anaconda myself). Could you try running your program again with adding 
`import koogu`
before you do the other 'from koogu import ...'?

**Comment by carlybatist** [comment]
Still doesn't work if I do 
`import koogu`
`from koogu import train, assessments, recognize`
`from koogu.data import preprocess, feeder`
`from koogu.model import architectures`

<img width="384" alt="Screen Shot 2022-08-13 at 8 33 47 AM" src="https://user-images.githubusercontent.com/46716228/184500874-e3c1ae10-0346-4e94-8945-f01103057bbb.png">

**Comment by shyamblast** [comment]
Can you provide the necessary details (platform, OS, Python version, pip/conda version, etc.) so I could try to reproduce this on my end? What version of TensorFlow is installed?

Also, as a quick "bandaid", could you possibly try replacing the line
`from koogu import train, assessments, recognize`
with
`from koogu.train_and_eval import train_and_eval as train`
`from koogu.utils import assessments`
`from koogu.inference import recognize`
and let me know if that resolves the hurdle?

**Comment by carlybatist** [comment]
I get the same 'koogu is not a package' error if I try your bandaid solution.

So it does work when I try locally on my laptop's terminal - MacOS version 12.5 (Monterey), laptop has Apple's M1 chip, using python version 3.8.13, pip 22.2, conda 4.12.0.  

Where it has not been working is while I am SSH'd into a virtual machine on Microsoft Azure using python version 3.7.13, pip 22.1.2, conda 4.13.0, with NVIDIA GPU (Tesla V100-PCIE), CUDA version 11.4. 

I don't have tensorflow installed as I am using Pytorch for my models, but wanted to just use Koogu's spectrogram/audio processing functions, which should not be dependent on TensorFlow..?

**Comment by shyamblast** [comment]
Ah, that explains the issue. The first screenshot you posted possibly didn't include the full output, so I was trying to guess things here- I believe towards the end of all those messages it should have complained that tensorflow wasn't available.
Short story is that you do need to have TensorFlow (TF) installed. It's marked as a soft dependency (since you could either install tensorflow or tensorflow-gpu, both of which expose similar API). So, `pip install koogu` should have displayed an error at the end. If it didn't, then I'll need to check why (please let me know).

Please do install TF before you install koogu. Even though you only want to use it for spectrogram/audio, the package's installation expects TF to be available.

**Comment by carlybatist** [comment]
pip install koogu did not display an error and didn't say anything about tensorflow needing to be installed. This is what it says in full. 

<img width="1024" alt="Screen Shot 2022-08-13 at 2 25 12 PM" src="https://user-images.githubusercontent.com/46716228/184511192-6a90742c-6029-4822-9f81-6327f1c13b1b.png">

On another note - I did just try installing tensorflow (successfully) and tried to re-install koogu and re-run the code importing it and I got the same error about koogu not being a package. So whatever is going on doesn't seem to be related to tensorflow being installed.

**Comment by shyamblast** [comment]
One more thing. Could you try explicitly uninstalling koogu and then installing fresh (once tensorflow is also installed)? I'm guessing that maybe your re-installing attempt didn't properly (re-)run the "post-install" instructions.

**Comment by NickleDave** [comment]
@carlybatist if your script is named `koogu` then it may be clobbering this package in Python's namespace.  

See for example this person with your same error:  
https://stackoverflow.com/questions/53749843/python-module-and-package-having-the-same-name

You're trying to import from the `koogu` you installed, but Python thinks you're trying to import the script itself, which takes precedence in the path:  
https://docs.python.org/3/tutorial/modules.html#the-module-search-path
> The directory containing the input script (or the current directory when no file is specified).
 
To quickly test if this fixes it, try renaming the script something else, e.g., `test_koogu.py`

To see what's going on more directly, insert a breakpoint in your script just before the statement that crashes in your script, run it, and then inspect the `__file__` attribute of `koogu`, like so:

your script:  
```
import pdb;pdb.set_trace()  #  `breakpoint()` in python >= 3.7
from koogu.data import feeder  # etc
```

then in the debugger you get from the breakpoint when you run I think you will find it's the path to your script.
```
(Pdb) koogu.__file__
cv4e/koogu.py
```

short version: don't name your script the same thing as the package you're using

Also as a side note you can use triple backticks as code fences and paste code in directly instead of screenshots. Makes it easier for maintainers to copy your error messages and Google them :slightly_smiling_face:

**Comment by carlybatist** [comment]
I changed the file name and am still getting the same errors. I'm not sure how to implement your breakpoint test and debugger?
I did:
`import pdb;pdb.set_trace()`
`breakpoint()`

and got
```
> /datadrive/Downloads/cv4e/test_koogu.py(2)<module>()
-> breakpoint()
(Pdb) 
[1]+  Stopped                 python cv4e/test_koogu.py
```
Apologies for my naivety, I'm a python newbie!

**Comment by NickleDave** [comment]
No worries and I hope I'm not leading you astray because I'm misunderstanding what's going on. If so sorry for adding noise to the issue

Did you add two lines?
```
import pdb;pdb.set_trace()
breakpoint()
```
You only need to use one of those -- `breakpoint()` is an alias for the more wordy `import pdb;pdb.set_trace()` that was added in Python 3.7 (that's what I was trying to indicate with my comment)

If you get the prompt `(Pdb)` then you should be able to type commands into it.  
Try typing `(Pdb) koogu.__file__` (without the `(Pdb)` part) and see what you get.

You can leave the debugger by typing `q` (more commands here: https://realpython.com/python-debugging-pdb/)

**Comment by shyamblast** [comment]
> 

@carlybatist did you get a chance to try this yet?

**Comment by shyamblast** [comment]
> One more thing. Could you try explicitly uninstalling koogu and then installing fresh (once tensorflow is also installed)? I'm guessing that maybe your re-installing attempt didn't properly (re-)run the "post-install" instructions.

@carlybatist did you get a chance to try this yet?

**Comment by shyamblast** [comment]
Closing due to no activity, and since issue couldn't be reproduced.
